src/functions/JpowerIntegrals.h

```diff
@@ -0,0 +1,152 @@
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#pragma once
+
+#include <iostream>
+#include <complex>
+#include <vector>
+#include <cmath>
+
+namespace mrcpp {
+
+/** @class JpowerIntegrals
+ *
+ * @brief A class needed for construction Schrodinger time evolution operator
+ *
+ * @details A two dimensional array consisting of integrals $J_m$ as follows.
+ * Our main operator has the following expansion
+\begin{equation}
+\label{operator_correlation_expansion}
+    \left[ \sigma_l^{\mathfrak n} \right]_{pj}
+    (a)
+    =
+    \sum_{k = 0}^{\infty}
+    C_{jp}^{2k}
+    \widetilde J_{2k + j + p}(l, a)
+    ,
+\end{equation}
+where $a = t \mathfrak N^2 = t 4^{\mathfrak n}$
+and
+\begin{equation}
+\label{power_integral}
+    \widetilde J_m
+    =
+    \frac
+    {
+        I_m
+        e^{ i \frac {\pi}4 (m - 1) }
+    }
+    {
+        2 \pi ( m + 2 )!
+    }
+    =
+    \frac
+    {
+        e^{ i \frac {\pi}4 (m - 1) }
+    }
+    {
+        2 \pi ( m + 2 )!
+    }
+    \int_{\mathbb R}
+    \exp
+    \left(
+        \rho l \exp \left( i \frac \pi 4 \right) - a \rho^2
+    \right)
+    \rho^m
+    d \rho
+\end{equation}
+satisfying the following relation
+\begin{equation}
+\label{power_integral_recursion}
+    \widetilde J_{m+1}
+    =
+    \frac
+    {
+        il
+    }
+    {
+        2a (m + 3)
+    }
+    \widetilde J_m
+    +
+    \frac {im}{2a(m + 2)(m + 3)}
+    \widetilde J_{m-1}
+    =
+    \frac
+    {
+        i
+    }
+    {
+        2a (m + 3)
+    }
+    \left(
+        l
+        \widetilde J_m
+        +
+        \frac {m}{(m + 2)}
+        \widetilde J_{m-1}
+    \right)
+    , \quad
+    m = 0, 1, 2, \ldots,
+\end{equation}
+with $\widetilde J_{-1} = 0$ and
+\begin{equation}
+\label{power_integral_0}
+    \widetilde J_0
+    =
+    \frac{ e^{ -i \frac{\pi}4 } }{ 4 \sqrt{ \pi a } }
+    \exp
+    \left(
+        \frac{il^2}{4a}
+    \right)
+    .
+\end{equation}
+ *
+ *  
+ */
+class JpowerIntegrals
+{
+public:
+    /// @brief creates an array of power integrals
+    /// @param a : parameter a
+    /// @param scaling : scaling level
+    /// @param M : maximum amount of integrals for each l
+    /// @param threshold : lower limit for neglecting the integrals
+    /// @details The array is orginised as a vector ordered as \f$l = 0, 1, 2, \ldots, 2^n - 1, 1 - 2^n, 2 - 2^n, \ldots, -2, -1 \f$.
+    JpowerIntegrals(double a, int scaling, int M, double threshold = 1.0e-15);
+    //JpowerIntegrals(const JpowerIntegrals& other);
+
+    //~JpowerIntegrals();
+
+    int scaling;  //it is probably not used
+    std::vector<std::vector<std::complex<double>>> integrals;
+
+    std::vector<std::complex<double>> & operator[](int index);
+private:
+    std::vector<std::complex<double>> calculate_J_power_integrals(int l, double a, int M, double threshold);
```

Comment:
Please use the naming convention for class methods. In this case `calculateJpowerIntegrals`

Reply:
This is indeed, one of the very few examples, where I do not follow the usual mrcpp convention. At leat somewhere in the code I would like to use those notations that I can read myself. It makes me upset, when I cannot read my own code. It also slows me down a lot. This piece `calculate_J_power_integrals` I can read and grasp in less than a second. This piece `calculateJpowerIntegrals` I need to process in my head for 10-20 seconds at least.

We already discussed it, but I remind again, sorry. It is not even possible to guess the meaning of this piece `int kp1_d = node.getKp1_d();`, though it is probably in line with the convention, or is it really in line?

As I said, mostly I follow the convention, and if somewhere I don't, it means that I want to be able to read the code I am writing. I am not a good programmer, and so I am not able to grasp things like this  `calculateJpowerIntegrals` or like this `getKp1_d`.